test/httpbis/httpbis.spec.ts

```diff
@@ -268,6 +286,31 @@ describe('httpbis', () => {
                     'value',
                     'value2',
                 ]);
+                expect(httpbis.deriveComponent('@query-param', new Map([['name', 'param']]), {
+                    ...req,
+                    url: new URL('https://example.com/path?param=value%7D&param=value2%7D'),
+                })).to.deep.equal([
+                    'value%7D',
+                    'value2%7D',
+                ]);
+                expect(httpbis.deriveComponent('@query-param', new Map([['name', 'var']]), {
+                    ...req,
+                    url: new URL('https://example.com/parameters?var=this%20is%20a%20big%0Amultiline%20value&bar=with+plus+whitespace&fa%C3%A7ade%22%3A%20=something'),
+                })).to.deep.equal([
+                    'this%20is%20a%20big%0Amultiline%20value',
+                ]);
+                expect(httpbis.deriveComponent('@query-param', new Map([['name', 'bar']]), {
+                    ...req,
+                    url: new URL('https://example.com/parameters?var=this%20is%20a%20big%0Amultiline%20value&bar=with+plus+whitespace&fa%C3%A7ade%22%3A%20=something'),
+                })).to.deep.equal([
+                    'with%20plus%20whitespace',

```

Comment:
OK - so this is interesting, it's actually not returning a data that was originally in the query string, which means there will be problems when computing signatures.

It seems that once the query string is parsed and stored in the `searchParams` prop, it is fully URL decoded and there's no way to get your hands on the raw value(s).

The only way to get the raw value is using `url.search` but that will need to be split and parsed into parts, which is going to be a bit of a pain.

Reply:
I took examples from the spec and I can't see any problems with the logic in my code: https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-message-signatures#name-query-parameters

Reply:
Please, note, I'm using `decodeURIComponent` instead of `decodeURI` which apparently does the trick with different representation as part of full URL or simple string

Reply:
https://stackoverflow.com/a/747700

Reply:
Ah! Great, I hadn't realised the spec specifically wanted this decode/recode logic. I think the spec here is likely to cause some clients to struggle to be compliant, but that's not our problem!